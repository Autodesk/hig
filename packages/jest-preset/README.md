# @hig/jest-preset

Jest preset for HIG packages.

## Usage

_Example `package.json`:_

```json
{
  "jest": {
    "preset": "@hig/jest-preset"
  }
}
```
