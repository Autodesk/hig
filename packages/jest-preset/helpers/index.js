/* eslint-disable global-require */
module.exports = {
  takeSnapshotsOf: require("./takeSnapshotsOf")
};
