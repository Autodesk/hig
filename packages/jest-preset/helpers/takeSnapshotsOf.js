const React = require("react");
const renderer = require("react-test-renderer");

module.exports = function takeSnapshotsOf(Component, cases, rendererOptions) {
  cases.forEach(({ desc, props }) => {
    it(desc, () => {
      const vDom = React.createElement(Component, props);
      const tree = renderer.create(vDom, rendererOptions).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
};
