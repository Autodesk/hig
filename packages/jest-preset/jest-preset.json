{
  "collectCoverageFrom": ["**/src/**/*.{js,jsx}"],
  "coverageDirectory": "<rootDir>/coverage",
  "coveragePathIgnorePatterns": [
    "__gemini__",
    "__stories__",
    "docs",
    "examples/redux",
    "node_modules",
    "packages/jest-preset",
    "packages/vanilla",
    "src/playground"
  ],
  "coverageThreshold": {
    "global": {
      "branches": 50,
      "functions": 50,
      "lines": 50,
      "statements": 50
    }
  },
  "moduleNameMapper": {
    "\\.(css|scss|svg)$": "@hig/jest-preset/fileMock.js"
  },
  "modulePaths": ["node_modules"],
  "moduleFileExtensions": ["js", "jsx", "json"],
  "setupFiles": ["raf/polyfill"],
  "setupTestFrameworkScriptFile": "@hig/jest-preset/setupTests.js",
  "testPathIgnorePatterns": [
    "docs",
    "packages/babel-preset",
    "packages/jest-preset",
    "packages/vanilla"
  ],
  "unmockedModulePathPatterns": ["node_modules"]
}
