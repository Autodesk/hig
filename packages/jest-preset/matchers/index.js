require("./toBeConstants");
require("./toImplementHIGInterfaceOf");
require("./toHavePropTypesAndDocgenInfo");
