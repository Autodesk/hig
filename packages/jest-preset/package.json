{
  "private": true,
  "name": "@hig/jest-preset",
  "version": "0.1.0",
  "devDependencies": {
    "babel-core": "^6.26.3",
    "babel-jest": "^23.4.0",
    "enzyme": "^2.9.1",
    "jest": "^21.2.1",
    "jest-enzyme": "^3.8.3",
    "jest-extended": "^0.6.0",
    "raf": "^3.4.0",
    "react-test-renderer": "^15.6.1"
  },
  "eslintConfig": {
    "extends": "@hig"
  },
  "jest": {
    "preset": "@hig/jest-preset"
  }
}
