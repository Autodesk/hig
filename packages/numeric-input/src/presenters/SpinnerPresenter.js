import React, { Component } from "react";
import PropTypes from "prop-types";
import { css } from "emotion";
import { ThemeContext } from "@hig/theme-context";
import stylesheet from "./stylesheet";
import { CaretUpMUI, CaretUpSUI, CaretDownMUI, CaretDownSUI } from "@hig/icons";

export default class SpinnerPresenter extends Component {
    static propTypes = {
        value: PropTypes.number,
        disabled: PropTypes.bool,
        hasFocus: PropTypes.bool,
        hasHover: PropTypes.bool,
        isPressed: PropTypes.bool,
        onBlur: PropTypes.func,
        onClick: PropTypes.func,
        onFocus: PropTypes.func,
        onMouseDown: PropTypes.func,
        onMouseEnter: PropTypes.func,
        onMouseLeave: PropTypes.func,
        onMouseUp: PropTypes.func
    }

    static defaultProps = {
        disabled: false,
        value: "0"
    }

    render() {
        const {
            value,
            disabled,
            hasFocus,
            hasHover,
            isPressed,
            onBlur,
            onClick,
            onChange,
            onMouseDown,
            onMouseEnter,
            onMouseLeave,
            onMouseUp,
            onFocus,
            increment,
            decrement,
            ...otherProps
        } = this.props;
        return (
            <ThemeContext.Consumer>
                {({ resolvedRoles, metadata }) => {
                    const styles = stylesheet (
                        {
                            disabled,
                            hasFocus,
                            hasHover,
                            isPressed
                        }, 
                        resolvedRoles
                    );
                    const UpIcon = metadata.densityId === "medium-density" ? CaretUpMUI : CaretUpSUI;
                    const DownIcon = metadata.densityId === "medium-density" ? CaretDownMUI : CaretDownSUI;
                    const ifKeyIsEnter = action => event => {
                        if (event.key === "Enter") {
                          action();
                        }
                      };
                    return (
                        <div className={css(styles.spinnerWrapper)}>
                            <span 
                                onBlur={onBlur}
                                onClick={increment}
                                onFocus={onFocus}
                                onChange={onChange}
                                onMouseDown={onMouseDown}
                                onMouseEnter={onMouseEnter}
                                onMouseLeave={onMouseLeave}
                                onMouseUp={onMouseUp}
                                onKeyDown={ifKeyIsEnter(increment)}
                                className = {css(styles.spinner)}
                                >
                                    <UpIcon/>
                            </span>
                            <span 
                                onBlur={onBlur}
                                onClick={decrement}
                                onFocus={onFocus}
                                onChange={onChange}
                                onMouseDown={onMouseDown}
                                onMouseEnter={onMouseEnter}
                                onMouseLeave={onMouseLeave}
                                onMouseUp={onMouseUp}
                                onKeyDown={ifKeyIsEnter(decrement)}
                                className = {css(styles.spinner)}
                                >
                                    <DownIcon/>
                            </span>
                        </div> 

                    )
                }}
            </ThemeContext.Consumer>
        )
    }
}

                        /*
                        */
/*
                        
        
        
        <span className={css(styles.spinnerWrapper)}>
                            <span >
                                <CaretUp16 color={iconColor}/>
                            </span>
                            <span className={css(styles.spinnerDown)}>
                                <CaretDown16 />
                            </span>
                        </span>
                        
                        
                            <span>
                                <input
                                    className={css(styles.spinner1)}
                                    disabled={disabled}
                                    onBlur={onBlur}
                                    onClick={onClick}
                                    onFocus={onFocus}
                                    onMouseDown={onMouseDown}
                                    onMouseEnter={onMouseEnter}
                                    onMouseLeave={onMouseLeave}
                                    onMouseUp={onMouseUp}
                                    type="button"
                                    {...otherProps}
                                />
                                <UpPresenter
                                    disabled={disabled}
                                    hasFocus={hasFocus}
                                    hasHover={hasHover}
                                    isPressed={isPressed}
                                />
                            </span>
                            <span>
                                <input
                                        className={css(styles.spinner2)}
                                        disabled={disabled}
                                        onBlur={onBlur}
                                        onClick={onClick}
                                        onFocus={onFocus}
                                        onMouseDown={onMouseDown}
                                        onMouseEnter={onMouseEnter}
                                        onMouseLeave={onMouseLeave}
                                        onMouseUp={onMouseUp}
                                        type="button"
                                        {...otherProps}
                                    />
                                    <DownPresenter
                                        disabled={disabled}
                                        hasFocus={hasFocus}
                                        hasHover={hasHover}
                                        isPressed={isPressed}
                                    />
                            </span>
                            */