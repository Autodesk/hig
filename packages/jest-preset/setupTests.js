require("jest-enzyme");
require("jest-extended");
require("./matchers");
