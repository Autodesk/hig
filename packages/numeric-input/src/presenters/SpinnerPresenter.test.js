import { takeSnapshotsOf } from "@hig/jest-preset/helpers";

import SpinnerPresenter from "./SpinnerPresenter";

describe("spinner/SpinnerPresenter/SpinnerPresenter", () => {
  takeSnapshotsOf(SpinnerPresenter, [
    {
      desc: "renders without props",
      props: {}
    },
    {
      desc: "renders with disabled",
      props: {
        disabled: true
      }
    },
    {
      desc: "renders with variant",
      props: {
        variant: "box"
      }
    },
    {
      desc: "renders with custom stylesheet",
      props: {
        stylesheet: styles => ({
          ...styles,
          spinner: {
            ...styles.spinner,
            backgroundColor: "pink"
          }
        })
      }
    },
    {
      desc: "renders with custom className",
      props: {
        className: "custom"
      }
    }
  ]);
});
