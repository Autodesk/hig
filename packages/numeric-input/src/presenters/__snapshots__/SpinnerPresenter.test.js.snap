// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`spinner/SpinnerPresenter renders with custom className 1`] = `
.emotion-5 {
  box-sizing: border-box;
  position: absolute;
  z-index: 1;
  line-height: 10px;
  cursor: pointer;
}

.emotion-4 {
  height: calc(36px - 2px);
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: calc((36px - 2px) / 2);
  outline: none;
  box-sizing: border-box;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
}

.emotion-1:first-of-type {
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.emotion-1 svg * {
  fill: rgba(102,102,102,0.7);
}

.emotion-1:hover svg * {
  fill: #666666;
}

.emotion-1:active svg * {
  fill: #3c3c3c;
}

.emotion-0 {
  fill: #808080;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-0 > * {
  fill: #808080;
}

.emotion-2 {
  fill: #808080;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-2 > * {
  fill: #808080;
}

<div
  className="emotion-5 custom__spinner-wrapper"
>
  <div
    className="emotion-4 custom__box-wrapper"
  >
    <span
      className="emotion-1 custom__spinner"
      onClick={undefined}
      onKeyDown={undefined}
      onMouseDown={undefined}
      onMouseEnter={undefined}
      onMouseLeave={undefined}
      onMouseUp={undefined}
      role="button"
      tabIndex="-1"
    >
      <svg
        className="custom__spinner-icon-up emotion-0"
        height="10px"
        viewBox="0 0 10 10"
        width="10px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.71 6.78a.75.75 0 0 1-1.06 0L5 4.13 2.35 6.78a.75.75 0 0 1-1.06 0 .74.74 0 0 1 0-1.06l3.36-3.36a.5.5 0 0 1 .7 0l3.36 3.36a.74.74 0 0 1 0 1.06z"
        />
      </svg>
    </span>
    <span
      className="emotion-1 custom__spinner"
      onClick={undefined}
      onKeyDown={undefined}
      onMouseDown={undefined}
      onMouseEnter={undefined}
      onMouseLeave={undefined}
      onMouseUp={undefined}
      role="button"
      tabIndex="-1"
    >
      <svg
        className="custom__spinner-icon-down emotion-2"
        height="10px"
        viewBox="0 0 10 10"
        width="10px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.71 3.22a.75.75 0 0 0-1.06 0L5 5.87 2.35 3.22a.75.75 0 0 0-1.06 0 .74.74 0 0 0 0 1.06l3.36 3.36a.5.5 0 0 0 .7 0l3.36-3.36a.74.74 0 0 0 0-1.06z"
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`spinner/SpinnerPresenter renders with custom stylesheet 1`] = `
.emotion-5 {
  box-sizing: border-box;
  position: absolute;
  z-index: 1;
  line-height: 10px;
  cursor: pointer;
}

.emotion-4 {
  height: calc(36px - 2px);
}

.emotion-0 {
  fill: #808080;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-0 > * {
  fill: #808080;
}

.emotion-2 {
  fill: #808080;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-2 > * {
  fill: #808080;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: calc((36px - 2px) / 2);
  outline: none;
  box-sizing: border-box;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  background-color: pink;
}

.emotion-1:first-of-type {
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.emotion-1 svg * {
  fill: rgba(102,102,102,0.7);
}

.emotion-1:hover svg * {
  fill: #666666;
}

.emotion-1:active svg * {
  fill: #3c3c3c;
}

<div
  className="emotion-5"
>
  <div
    className="emotion-4"
  >
    <span
      className="emotion-1"
      onClick={undefined}
      onKeyDown={undefined}
      onMouseDown={undefined}
      onMouseEnter={undefined}
      onMouseLeave={undefined}
      onMouseUp={undefined}
      role="button"
      tabIndex="-1"
    >
      <svg
        className="emotion-0"
        height="10px"
        viewBox="0 0 10 10"
        width="10px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.71 6.78a.75.75 0 0 1-1.06 0L5 4.13 2.35 6.78a.75.75 0 0 1-1.06 0 .74.74 0 0 1 0-1.06l3.36-3.36a.5.5 0 0 1 .7 0l3.36 3.36a.74.74 0 0 1 0 1.06z"
        />
      </svg>
    </span>
    <span
      className="emotion-1"
      onClick={undefined}
      onKeyDown={undefined}
      onMouseDown={undefined}
      onMouseEnter={undefined}
      onMouseLeave={undefined}
      onMouseUp={undefined}
      role="button"
      tabIndex="-1"
    >
      <svg
        className="emotion-2"
        height="10px"
        viewBox="0 0 10 10"
        width="10px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.71 3.22a.75.75 0 0 0-1.06 0L5 5.87 2.35 3.22a.75.75 0 0 0-1.06 0 .74.74 0 0 0 0 1.06l3.36 3.36a.5.5 0 0 0 .7 0l3.36-3.36a.74.74 0 0 0 0-1.06z"
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`spinner/SpinnerPresenter renders with disabled 1`] = `
.emotion-0 {
  fill: #808080;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-0 > * {
  fill: #808080;
}

.emotion-2 {
  fill: #808080;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-2 > * {
  fill: #808080;
}

.emotion-5 {
  box-sizing: border-box;
  position: absolute;
  z-index: 1;
  line-height: 10px;
  cursor: default;
}

.emotion-4 {
  height: calc(36px - 2px);
  opacity: 0.4;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: calc((36px - 2px) / 2);
  outline: none;
  box-sizing: border-box;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
}

.emotion-1:first-of-type {
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.emotion-1 svg * {
  fill: rgba(102,102,102,0.7);
}

.emotion-1:hover svg * {
  fill: rgba(102,102,102,0.7);
}

.emotion-1:active svg * {
  fill: rgba(102,102,102,0.7);
}

<div
  className="emotion-5"
>
  <div
    className="emotion-4"
  >
    <span
      className="emotion-1"
      onClick={undefined}
      onKeyDown={undefined}
      onMouseDown={undefined}
      onMouseEnter={undefined}
      onMouseLeave={undefined}
      onMouseUp={undefined}
      role="button"
      tabIndex="-1"
    >
      <svg
        className="emotion-0"
        height="10px"
        viewBox="0 0 10 10"
        width="10px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.71 6.78a.75.75 0 0 1-1.06 0L5 4.13 2.35 6.78a.75.75 0 0 1-1.06 0 .74.74 0 0 1 0-1.06l3.36-3.36a.5.5 0 0 1 .7 0l3.36 3.36a.74.74 0 0 1 0 1.06z"
        />
      </svg>
    </span>
    <span
      className="emotion-1"
      onClick={undefined}
      onKeyDown={undefined}
      onMouseDown={undefined}
      onMouseEnter={undefined}
      onMouseLeave={undefined}
      onMouseUp={undefined}
      role="button"
      tabIndex="-1"
    >
      <svg
        className="emotion-2"
        height="10px"
        viewBox="0 0 10 10"
        width="10px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.71 3.22a.75.75 0 0 0-1.06 0L5 5.87 2.35 3.22a.75.75 0 0 0-1.06 0 .74.74 0 0 0 0 1.06l3.36 3.36a.5.5 0 0 0 .7 0l3.36-3.36a.74.74 0 0 0 0-1.06z"
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`spinner/SpinnerPresenter renders with variant 1`] = `
.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: calc((36px - 2px) / 2);
  outline: none;
  box-sizing: border-box;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
}

.emotion-1:first-of-type {
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.emotion-1 svg * {
  fill: rgba(102,102,102,0.7);
}

.emotion-1:hover svg * {
  fill: #666666;
}

.emotion-1:active svg * {
  fill: #3c3c3c;
}

.emotion-0 {
  fill: #808080;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-0 > * {
  fill: #808080;
}

.emotion-2 {
  fill: #808080;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-2 > * {
  fill: #808080;
}

.emotion-5 {
  box-sizing: border-box;
  position: absolute;
  z-index: 1;
  line-height: 10px;
  cursor: pointer;
  width: 35px;
  right: 0px;
  height: 100%;
}

.emotion-4 {
  height: 36px;
  z-index: -2;
  border-left-style: solid;
  border-left-color: rgba(60,60,60,0.1);
  border-left-width: 1px;
  margin-top: 1px;
  margin-bottom: 1px;
  margin-right: 1px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transform: translateY(-1px);
  -ms-transform: translateY(-1px);
  transform: translateY(-1px);
}

<div
  className="emotion-5"
>
  <div
    className="emotion-4"
  >
    <span
      className="emotion-1"
      onClick={undefined}
      onKeyDown={undefined}
      onMouseDown={undefined}
      onMouseEnter={undefined}
      onMouseLeave={undefined}
      onMouseUp={undefined}
      role="button"
      tabIndex="-1"
    >
      <svg
        className="emotion-0"
        height="10px"
        viewBox="0 0 10 10"
        width="10px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.71 6.78a.75.75 0 0 1-1.06 0L5 4.13 2.35 6.78a.75.75 0 0 1-1.06 0 .74.74 0 0 1 0-1.06l3.36-3.36a.5.5 0 0 1 .7 0l3.36 3.36a.74.74 0 0 1 0 1.06z"
        />
      </svg>
    </span>
    <span
      className="emotion-1"
      onClick={undefined}
      onKeyDown={undefined}
      onMouseDown={undefined}
      onMouseEnter={undefined}
      onMouseLeave={undefined}
      onMouseUp={undefined}
      role="button"
      tabIndex="-1"
    >
      <svg
        className="emotion-2"
        height="10px"
        viewBox="0 0 10 10"
        width="10px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.71 3.22a.75.75 0 0 0-1.06 0L5 5.87 2.35 3.22a.75.75 0 0 0-1.06 0 .74.74 0 0 0 0 1.06l3.36 3.36a.5.5 0 0 0 .7 0l3.36-3.36a.74.74 0 0 0 0-1.06z"
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`spinner/SpinnerPresenter renders without props 1`] = `
.emotion-5 {
  box-sizing: border-box;
  position: absolute;
  z-index: 1;
  line-height: 10px;
  cursor: pointer;
}

.emotion-4 {
  height: calc(36px - 2px);
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: calc((36px - 2px) / 2);
  outline: none;
  box-sizing: border-box;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
}

.emotion-1:first-of-type {
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.emotion-1 svg * {
  fill: rgba(102,102,102,0.7);
}

.emotion-1:hover svg * {
  fill: #666666;
}

.emotion-1:active svg * {
  fill: #3c3c3c;
}

.emotion-0 {
  fill: #808080;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-0 > * {
  fill: #808080;
}

.emotion-2 {
  fill: #808080;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-2 > * {
  fill: #808080;
}

<div
  className="emotion-5"
>
  <div
    className="emotion-4"
  >
    <span
      className="emotion-1"
      onClick={undefined}
      onKeyDown={undefined}
      onMouseDown={undefined}
      onMouseEnter={undefined}
      onMouseLeave={undefined}
      onMouseUp={undefined}
      role="button"
      tabIndex="-1"
    >
      <svg
        className="emotion-0"
        height="10px"
        viewBox="0 0 10 10"
        width="10px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.71 6.78a.75.75 0 0 1-1.06 0L5 4.13 2.35 6.78a.75.75 0 0 1-1.06 0 .74.74 0 0 1 0-1.06l3.36-3.36a.5.5 0 0 1 .7 0l3.36 3.36a.74.74 0 0 1 0 1.06z"
        />
      </svg>
    </span>
    <span
      className="emotion-1"
      onClick={undefined}
      onKeyDown={undefined}
      onMouseDown={undefined}
      onMouseEnter={undefined}
      onMouseLeave={undefined}
      onMouseUp={undefined}
      role="button"
      tabIndex="-1"
    >
      <svg
        className="emotion-2"
        height="10px"
        viewBox="0 0 10 10"
        width="10px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.71 3.22a.75.75 0 0 0-1.06 0L5 5.87 2.35 3.22a.75.75 0 0 0-1.06 0 .74.74 0 0 0 0 1.06l3.36 3.36a.5.5 0 0 0 .7 0l3.36-3.36a.74.74 0 0 0 0-1.06z"
        />
      </svg>
    </span>
  </div>
</div>
`;
